pyccel/parser/semantic.py

```diff
@@ -1116,11 +1138,37 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign,ar
             if not arr_in_multirets:
                 self._ensure_target(rhs, d_lhs)
 
-            var = self.check_for_variable(name)
+            var = self.check_for_variable(lhs)
 
             # Variable not yet declared (hence array not yet allocated)
             if var is None:
 
+                if isinstance(lhs, DottedName):
+                    prefix_parts = lhs.name[:-1]
+                    syntactic_prefix = prefix_parts[0] if len(prefix_parts) == 1 else DottedName(*prefix_parts)
+                    prefix = self._visit(syntactic_prefix)
+                    class_def = prefix.cls_base
+                    if isinstance(class_def, ClassDef) and lhs.name[0] == 'self':
```

Comment:
Is it possible for `class_def` to be a type other than `ClassDef`? If so then the associated error doesn't seem to be handled.

I suspect that `class_def` can only be a `ClassDef`. It may be `None` if there is a bug in our semantics but this probably doesn't need to be caught.
If you agree then I think we can remove this part of the condition

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1484/commits/e7b19d20720e2c2800a7858162928c0aae6ea639.